diffusion_with_autoscaler/autoscaler.py

```diff
@@ -231,49 +232,81 @@ async def send_batch(self, batch: List[Tuple[str, _BatchRequestModel]], server_u
+                async with self._lock:
+                    if self.batching == "streamed":
+                        # decrement the number of currently processed requests.
+                        self._server_status[server_url] -= 1
+                    else:
+                        # TODO - if the server returns an error, track that so
+                        #  we don't send more requests to it
+                        self._server_status[server_url] = True
 
     def _find_free_server(self) -> Optional[str]:
         existing = set(self._server_status.keys())
         for server in existing:
             status = self._server_status.get(server, None)
-            if status is None:
-                logger.error("Server is not found in the status list. This should not happen.")
-            if status:
-                return server
+            if self.batching == "streamed":
+                if status is None:
+                    logger.error("Server is not found in the status list. This should not happen.")
+                # Return server only if the status is strictly lower than the total number of requests.
+                if status < self.max_batch_size:
+                    return server
+            else:
+                if status is None:
+                    logger.error("Server is not found in the status list. This should not happen.")
+                if status:
+                    return server
 
     async def consumer(self):
         """The consumer process that continuously checks for new requests and sends them to the API.
 
         Two instances of this function should not be running with shared `_state_server` as that would create race
         conditions
         """
-        while True:
-            await asyncio.sleep(0.05)
-            batch = self._batch[: self.max_batch_size]
-            is_batch_ready = len(batch) == self.max_batch_size
-            if len(batch) > 0 and self._last_batch_sent is None:
-                self._last_batch_sent = time.time()
-
-            if self._last_batch_sent:
-                is_batch_timeout = time.time() - self._last_batch_sent > self.timeout_batching
-            else:
-                is_batch_timeout = False
-
-            server_url = self._find_free_server()
-            # setting the server status to be busy! This will be reset by
-            # the send_batch function after the server responds
-            if server_url is None:
-                continue
-            if batch and (is_batch_ready or is_batch_timeout):
-                self._server_status[server_url] = False
-                # find server with capacity
-                asyncio.create_task(self.send_batch(batch, server_url))
-                # resetting the batch array, TODO - not locking the array
-                self._batch = self._batch[len(batch) :]
-                self._last_batch_sent = time.time()
+        try:
+            while True:
+                await asyncio.sleep(0.05)
```

Comment:
it's ok to sleep here but we don't have to, strictly speaking

Reply:
but let's keep it in, maybe we can take it down to 0.01 (have you tried different values?)

Reply:
Changed to 0.001